### PR TITLE
Perspective beta 0.7

### PR DESCRIPTION
**Pull request by stuartmscott**
- Store solutions under puzzle hash
- Update Gradle
- Add World 13
- Make all puzzles available in debug

### CODE REVIEW

File: app/src/main/java/com/aletheiaware/perspective/android/ui/GameActivity.java [hunk 1]
```diff
@@ -431,6 +435,10 @@ private String getNextWorld() {
                 if (manager.hasPurchased(PerspectiveAndroidUtils.WORLD_TWELVE)) {
                     return PerspectiveAndroidUtils.WORLD_TWELVE;
                 } // else fallthrough
+            case PerspectiveAndroidUtils.WORLD_TWELVE:
+                if (manager.hasPurchased(PerspectiveAndroidUtils.WORLD_THIRTEEN)) {
+                    return PerspectiveAndroidUtils.WORLD_THIRTEEN;
+                } // else fallthrough
             default:
                 return null;
```
Comment (winksaville) [inline app/src/main/java/com/aletheiaware/perspective/android/ui/GameActivity.java:443]: FTF: Suggestion for the future return an empty string or throw an exception and then mark routine as @NonNull.

File: app/src/main/java/com/aletheiaware/perspective/android/ui/WorldSelectActivity.java [hunk 2]
```diff
@@ -131,7 +150,7 @@ public void onSelect(World world) {
             PuzzleAdapter puzzleAdapter = new PuzzleAdapter(this, world, puzzleStars.get(name), new PuzzleAdapter.Callback() {
                 @Override
                 public void onSelect(String world, int puzzle) {
-                    if (dialog != null && dialog.isShowing()) {
+                    if (dialog.isShowing()) {
```
Comment (winksaville) [inline app/src/main/java/com/aletheiaware/perspective/android/ui/WorldSelectActivity.java:153]: FTF: I'd like to suggest that you strive to use @NonNull/@Nullable as much as possible and minimize methods that return null. For example, see comment on getNextWorld.

File: app/src/androidTest/java/com/aletheiaware/perspective/android/GameActivityInstrumentedTest.java [hunk 3]
```diff
@@ -259,6 +259,11 @@ public void screenshotWorld1Puzzle6() throws Exception {
         captureScreenshot(PerspectiveAndroidUtils.WORLD_ONE, 6);
     }
 
+    @Test
+    public void screenshotWorld1Puzzle7() throws Exception {
+        captureScreenshot(PerspectiveAndroidUtils.WORLD_ONE, 7);
```
Comment (winksaville) [inline app/src/androidTest/java/com/aletheiaware/perspective/android/GameActivityInstrumentedTest.java:264]: For The Future (FTF): I'm guessing you manually review the screen captures, any possible way to "validate" these at test time?

**Review by winksaville — COMMENTED** [review]
LGTM, a few neophyte observations, see the line comments.

Hmmm, I don't see away to "Approve" since I have comments, consider it so. Of course you can mark them resolved and I'll formally approve.

File: app/src/main/java/com/aletheiaware/perspective/android/ui/GameActivity.java (hunk 1 above)
Comment (stuartmscott) [reply app/src/main/java/com/aletheiaware/perspective/android/ui/GameActivity.java:443]: Done in Beta 0.8

File: app/src/main/java/com/aletheiaware/perspective/android/ui/WorldSelectActivity.java (hunk 2 above)
Comment (stuartmscott) [reply app/src/main/java/com/aletheiaware/perspective/android/ui/WorldSelectActivity.java:153]: Done

File: app/src/androidTest/java/com/aletheiaware/perspective/android/GameActivityInstrumentedTest.java (hunk 3 above)
Comment (stuartmscott) [reply app/src/androidTest/java/com/aletheiaware/perspective/android/GameActivityInstrumentedTest.java:264]: The puzzles themselves are validated before being committed to PerspectivePuzzle. These screenshot tests are review manually for consistency before the puzzles are play tested.

**Pull request merged**